### ISSUE TITLE
In the case of bean circular dependency, the springAOP annotation @SchedulerLock is invalid.

### ISSUE THREAD

**Issue by LiveInTheMoment** [lukas-krecan/ShedLock#2061]
Dear author，
1. I have read that connecting (https://github.com/lukas-krecan/ShedLock#troubleshooting) does not solve the problem
2. I've tried upgrading to the latest version, but it doesn't work either
So I need to report this issue to you and hope to get a reply, thank you。
[Details]:
version:4.36.0 

I will use simple examples to show you better:
------------------------------

```java
public interface ServiceQ {
    void performTask();
}
import net.javacrumbs.shedlock.core.LockAssert;
import net.javacrumbs.shedlock.spring.annotation.SchedulerLock;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Service;

@Service
public class ServiceQImpl implements ServiceQ {

    @Autowired
    private ServiceZ serviceZ;

    @Override
    @Scheduled(fixedRate = 5000)
    @SchedulerLock(name = "myScheduledTask1", lockAtMostFor = "PT5S", lockAtLeastFor = "PT5S")
    public void performTask() {
        LockAssert.assertLocked();
        System.out.println("Q Performing task...");
    }

}

public interface ServiceZ {
    void performTask();
}
@Service
public class ServiceZImpl implements ServiceZ {

    @Autowired
    private ServiceQ serviceQ;

    @Override
    public void performTask() {

    }

}

@SpringBootApplication
@EnableAspectJAutoProxy
public class Application {
    public static void main(String[] args) {
        ApplicationContext context = SpringApplication.run(Application.class, args);
		}
		}
```
------------------------------

As shown in the code.serviceQ and serviceZ are circular dependencies，Error after startup
```
[2024-08-02 10:20:20,205][scheduling-1][ERROR][org.springframework.scheduling.support.TaskUtils$LoggingErrorHandler.handleError(TaskUtils.java:95)] - Unexpected error occurred in scheduled task
java.lang.IllegalStateException: The task is not locked.
        at net.javacrumbs.shedlock.core.LockAssert.assertLocked(LockAssert.java:49) ~[shedlock-core-4.44.0.jar:?]
        at com.example.demo.sven4.ServiceQImpl.performTask(ServiceQImpl.java:27) ~[classes/:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_121]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_121]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_121]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_121]
        at org.springframework.scheduling.support.ScheduledMethodRunnable.run(ScheduledMethodRunnable.java:84) ~[spring-context-5.3.34.jar:5.3.34]
        at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54) ~[spring-context-5.3.34.jar:5.3.34]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_121]
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) ~[?:1.8.0_121]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) ~[?:1.8.0_121]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) ~[?:1.8.0_121]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[?:1.8.0_121]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[?:1.8.0_121]
        at java.lang.Thread.run(Thread.java:745) ~[?:1.8.0_121]
```

**Comment by lukas-krecan** [comment]
Hi, I tried to reproduce it and I got from Spring

> Relying upon circular references is discouraged and they are prohibited by default.

I think it speeks for itself. But I wanted to see it for myself and tired to reproduce it [here](https://github.com/lukas-krecan/ShedLock/compare/circular?expand=1) and it works. Even if I managed to reproduce it, I couldn't do much about it as ShedLock relies on Spring AOP and if it does not work in your case, the reason is most likely that Spring AOP fails to work on circular dependencies.
